### PR TITLE
(maint) Add yard documentation

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

File: lib/facter.rb [hunk 1]
```diff
@@ -47,65 +76,104 @@ def on_message(&block)
       Facter::Log.on_message(&block)
     end
 
+    # Check whether debuging is inabled
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: enabled is spelled incorrectly

File: lib/facter.rb [hunk 2]
```diff
@@ -47,65 +76,104 @@ def on_message(&block)
...
+    # @return [void]
+    #
+    # @api public
     def search_external(dirs)
       LegacyFacter.search_external(dirs)
     end
 
+    # Returns the registered search directories.
+    #
+    # @return [Array<String>] An array of the directories searched
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: An array of searched directories, sounds better

File: lib/facter.rb [hunk 3]
```diff
@@ -47,65 +76,104 @@ def on_message(&block)
...
+    #
+    # @api public
     def search_external_path
       LegacyFacter.search_external_path
     end
 
+    # Returns the registered search directories.
+    #
+    # @return [Array<String>] An array of the directories searched
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: Same as the above comment

File: lib/facter.rb [hunk 4]
```diff
@@ -132,14 +205,31 @@ def to_user_output(cli_options, *args)
       [fact_formatter.format(resolved_facts), status || 0]
     end
 
+    # Check whether printing stack trace is inabled
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: enabled spelled incorrectly

File: lib/facter.rb [hunk 5]
```diff
@@ -132,14 +205,31 @@ def to_user_output(cli_options, *args)
...
+    #
+    # @return [bool]
+    #
+    # @api public
     def trace?
       LegacyFacter.trace?
     end
 
+    # Enable or disable trace
+    # @param debug_bool [bool] State which trace should have
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: Set trace state, or Set trace on debug state, probably sounds better

File: lib/facter.rb [hunk 6]
```diff
@@ -132,14 +205,31 @@ def to_user_output(cli_options, *args)
...
+    #
+    # @api public
     def trace(bool)
       LegacyFacter.trace(bool)
     end
 
+    # Gets the value for a fact. Returns `nil` if no such fact exists.
+    #
+    # @param name [String] the fact name
+    # @return [String the value of the fact, or nil if no fact is found
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: missing ']' after String. The fact value, or nil if no fact is found, sounds better

File: lib/facter.rb [hunk 7]
```diff
@@ -150,18 +240,28 @@ def value(user_query)
...
-      @logger.debug("blocking collection of #{block_list.join("\s")} facts") if block_list.any? && Options[:block]
-    end
-
+    # Returns exit status when user query contains facts that do
+    #   not exist
+    #
+    # @param dirs [Array] Arguments sent to CLI
+    # @param dirs [Array] List of resolved facts
+    #
+    # @return [Integer, nil] Will return status 1 if ueser query contains
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: user spelled incorrectly

File: lib/facter.rb [hunk 8]
```diff
@@ -150,18 +240,28 @@ def value(user_query)
...
+    #  facts that are not found and resolved, otherwise it will return nil
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: Is it 'not found and resolved', or, 'not found or resolved'?

File: lib/facter.rb [hunk 9]
```diff
@@ -175,5 +275,50 @@ def error_check(args, resolved_facts)
 
       status
     end
+
+    # Prints out blocked facts before when to_hash or to_user_output
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: What you are trying to say is: Also prints blocked facts, when to_hash or to_user_output is called?

File: lib/facter.rb [hunk 10]
```diff
@@ -175,5 +275,50 @@ def error_check(args, resolved_facts)
...
+    #
+    # @return [nil]
+    #
+    # @api private
+    def log_blocked_facts
+      block_list = BlockList.instance.block_list
+      @logger.debug("blocking collection of #{block_list.join("\s")} facts") if block_list.any? && Options[:block]
+    end
+
+    # Print out error messages to
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: I think that Print an error message, sounds better

File: lib/facter.rb [hunk 11]
```diff
@@ -175,5 +275,50 @@ def error_check(args, resolved_facts)
...
+    # @return [nil]
+    #
+    # @api private
+    def log_errors(missing_names)
+      missing_names.each do |missing_name|
+        @logger.error("fact \"#{missing_name}\" does not exist.", true)
+      end
+    end
+
+    # Proxy method catch metod calls that were not implemented yet
```
Comment (Filipovici-Andrei) [inline lib/facter.rb]: method not spelled correctly and I think Proxy method that catches not yet implemented method calls, sounds a little better

**Review by Filipovici-Andrei — CHANGES_REQUESTED** [review]

File: lib/facter.rb (hunk 8 above)
Comment (sebastian-miclea) [reply lib/facter.rb]: it should be not found or resolved

File: lib/facter.rb [hunk 12]
```diff
@@ -53,65 +82,104 @@ def on_message(&block)
...
+    #
+    # @api public
     def reset
       LegacyFacter.reset
       LegacyFacter.search(*Options.custom_dir)
       LegacyFacter.search_external(Options.external_dir)
+      nil
     end
 
+    # Register directories to be searched for facts. The registered directories
```
Comment (IrimieBogdan) [inline lib/facter.rb]: ... to be searched for custom facts.

File: lib/facter.rb [hunk 13]
```diff
@@ -53,65 +82,104 @@ def on_message(&block)
...
+    # @param dirs [Array<String>] An array of searched directories
+    #
+    # @return [void]
+    #
+    # @api public
     def search_external(dirs)
       LegacyFacter.search_external(dirs)
     end
 
+    # Returns the registered search directories.
```
Comment (IrimieBogdan) [inline lib/facter.rb]: ... search directories for external facts.

File: lib/facter.rb [hunk 14]
```diff
@@ -53,65 +82,104 @@ def on_message(&block)
...
+    #
+    # @return [Array<String>] An array of searched directories
+    #
+    # @api public
     def search_external_path
       LegacyFacter.search_external_path
     end
 
+    # Returns the registered search directories.
```
Comment (IrimieBogdan) [inline lib/facter.rb]: ... search directories for custom facts.

File: lib/facter.rb [hunk 15]
```diff
@@ -53,65 +82,104 @@ def on_message(&block)
...
+    #
+    # @return [Array<String>] An array of the directories searched
+    #
+    # @api public
     def search_path
       LegacyFacter.search_path
     end
 
+    # Gets a hash mapping fact names to their values
```
Comment (IrimieBogdan) [inline lib/facter.rb]: The hash contains core facts, legacy facts, custom facts and external facts (all facts that can be resolved).

File: lib/facter.rb [hunk 16]
```diff
@@ -123,6 +191,11 @@ def to_hash
       FactCollection.new.build_fact_collection!(resolved_facts)
     end
 
+    # Gets a hash mapping fact names to their values
+    #
+    # @return [Array] the hash of fact names and values
+    #
+    # @api public
```
Comment (IrimieBogdan) [inline lib/facter.rb]: This is an internal API and must be private. It contains a string with facts formatted the way user specified from CLI.

File: lib/facter.rb [hunk 17]
```diff
@@ -198,5 +302,50 @@ def error_check(args, resolved_facts)
...
+    #
+    # @return [nil]
+    #
+    # @api private
+    def log_blocked_facts
+      block_list = BlockList.instance.block_list
+      @logger.debug("blocking collection of #{block_list.join("\s")} facts") if block_list.any? && Options[:block]
+    end
+
+    # I think that Print an error message
```
Comment (IrimieBogdan) [inline lib/facter.rb]: It is used for printing errors regarding CLI user input validation

File: lib/facter.rb [hunk 18]
```diff
@@ -164,6 +259,10 @@ def add_fact_to_searched_facts(user_query, value)
       @already_searched[user_query].value = value
     end
 
+    # Returns a fact object by name
```
Comment (IrimieBogdan) [inline lib/facter.rb]: Resolved facts and saves the result in @already_searched array that is used as a global collection.

File: lib/facter.rb (hunk 16 above)
Comment (sebastian-miclea) [reply lib/facter.rb]: This cannot be called as private, it gets called from outside the module.

**Review by BogdanIrimie — APPROVED** [review]

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**